src/udp.rs

```diff
@@ -82,6 +82,13 @@ impl UdpSocket {
 
     /// Receives data from the socket. On success, returns the number of bytes
     /// read and the address from whence the data came.
+    ///
+    /// The function must be called with valid byte array `buf` of sufficient size to
+    /// hold the message bytes. If a message is too long to fit in the supplied buffer,
+    /// excess bytes may be discarded.
+    ///
+    /// The function is overwriting previous content of `buf`. Slicing
+    /// `&buf[..NumRead]` allows efficient access and boundary checking at compile time.
```

Comment:
Assuming `NumRead` is the result of `recv_from` I think a more idiomatic name for it is `num_read`, so we know it's some local variable.